pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -216,7 +216,8 @@ def _do_check_save(self, filepath, current, epoch):
         if len(self.best_k_models) == self.save_top_k:
             delpath = self.kth_best_model
             self.best_k_models.pop(self.kth_best_model)
-            self._del_model(delpath)
+            if self.save_top_k > 0:
```

Comment:
@jeremyjordan this if shall be one level up, you don't want to pop (remove from list) model which you don't delete, right?